### ISSUE TITLE
"aws_apigatewayv2_integration" updates "passthrough_behavior" on every terraform plan

### ISSUE THREAD

**Issue by lxg** [terraform-providers/terraform-provider-aws#13893]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.26
+ provider.aws v2.67.0
+ provider.local v1.4.0
+ provider.null v2.1.2
+ provider.random v2.2.1
+ provider.template v2.1.2

### Affected Resource(s)

* aws_apigatewayv2_integration

### Terraform Configuration Files

Here’s the relevant code of our config:

```hcl
resource "aws_apigatewayv2_integration" "services_gateway" {
  api_id               = aws_apigatewayv2_api.services_gateway.id
  integration_type     = "HTTP_PROXY"
  connection_type      = "VPC_LINK"
  integration_method   = "ANY"
  passthrough_behavior = "WHEN_NO_MATCH"
  connection_id        = aws_apigatewayv2_vpc_link.services_gateway.id
  integration_uri      = module.tf_application_load_balancer.load_balancer_listener_arn
}
```

Here’s the output of `terraform plan`

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # aws_apigatewayv2_integration.services_gateway will be updated in-place
  ~ resource "aws_apigatewayv2_integration" "services_gateway" {
        api_id                 = "…"
        connection_id          = "…"
        connection_type        = "VPC_LINK"
        id                     = "…"
        integration_method     = "ANY"
        integration_type       = "HTTP_PROXY"
        integration_uri        = "arn:aws:elasticloadbalancing:…"
      + passthrough_behavior   = "WHEN_NO_MATCH"
        payload_format_version = "1.0"
        request_templates      = {}
        timeout_milliseconds   = 29000
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

### Expected Behavior

`terraform plan` should not schedule any changes. Especially as `passthrough_behavior = "WHEN_NO_MATCH"` is the default anyway. Removing the explicit setting in the config yields the same behaviour.

### Actual Behavior

This keeps happening after every terraform apply, so there is never a “consistent” state.

**Comment by ewbankkit** [comment]
@lxg Thanks for raising this issue.
It has been noticed previously and a fix submitted in #13062.

**Comment by lxg** [comment]
Oops, sorry for the duplicate. Missed that when searching existing issues. :(

**Comment by ewbankkit** [comment]
@lxg Actually, it's not a duplicate as the linked PR didn't have an explicit issue it was closing.
I'll reopen this one and it will be closed when the PR is merged.
Thanks.

**Comment by jdcookie** [comment]
Seeing the same issue - not defining `passthrough_behaviour` - it's empty in the terraform.tfstate but it shows up in the plan, adding it in, even though this is an HTTP endpoint and I believe they are only used for websockets?

`
resource "aws_apigatewayv2_integration" "AllProxy_integration" {
  api_id           = aws_apigatewayv2_api.config.id
  integration_type = "AWS_PROXY"

  connection_type      = "INTERNET"
  description          = "Lambda Integration"
  integration_method   = "POST"
  integration_uri      = aws_lambda_function.config.invoke_arn
}
`
`
{
      "mode": "managed",
      "type": "aws_apigatewayv2_integration",
      "name": "AllProxy_integration",
      "provider": "provider.aws",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "api_id": "...",
            "connection_id": "",
            "connection_type": "INTERNET",
            "content_handling_strategy": "",
            "credentials_arn": "",
            "description": "Lambda Integration",
            "id": "zfcgmh1",
            "integration_method": "POST",
            "integration_response_selection_expression": "",
            "integration_type": "AWS_PROXY",
            "integration_uri": "...",
            "passthrough_behavior": "",
            "payload_format_version": "1.0",
            "request_templates": {},
            "template_selection_expression": "",
            "timeout_milliseconds": 29000
          },
          "private": "bnVsbA==",
          "dependencies": [
            "aws_apigatewayv2_api.config",
            "aws_lambda_function.config"
          ]
        }
      ]
    }
`
`  ~ resource "aws_apigatewayv2_integration" "AllProxy_integration" {
        api_id                 = "..."
        connection_type        = "INTERNET"
        description            = "Lambda Integration"
        id                     = "..."
        integration_method     = "POST"
        integration_type       = "AWS_PROXY"
        integration_uri        = ...
      + passthrough_behavior   = "WHEN_NO_MATCH"
        payload_format_version = "1.0"
        request_templates      = {}
        timeout_milliseconds   = 29000
    }`

**Comment by ewbankkit** [comment]
Closed via #13062.
